### PR TITLE
Build pipepine path

### PR DESCRIPTION
**Pull request by exocom**
## All Submissions:
-------------------------------------
* [YES] Have you added an explanation of what your changes do and why you'd like us to include them?
* [YES] I have updated the documentation accordingly.
* [YES] I have added tests to cover my changes.
* [YES] All new and existing tests passed.
* [YES] My code follows the code style of this project.
* [YES] I ran lint checks locally prior to submission.
* [YES] Have you checked to ensure there aren't other open PRs for the same update/change?

## What is the current behavior?
-------------------------------------

When creating a build definition the build will be placed into the root directory.
Issue Number: N/A

## What is the new behavior?
-------------------------------------

- You can now specify the directory for the build pipeline to be placed into.
- If the directory does not exist it will automatically be created.
- If you do not specify a directory the pipeline will be placed into the root. Same as current behavior.

## Does this introduce a breaking change?
-------------------------------------
- [NO]

## Any relevant logs, error output, etc?
-------------------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

## Other information
-------------------------------------

### CODE REVIEW

File: azuredevops/resource_build_definition.go [hunk 1]
```diff
@@ -36,6 +36,11 @@ func resourceBuildDefinition() *schema.Resource {
 				Optional: true,
 				Default:  "",
 			},
+			"path": {
```
Comment (nmiodice) [inline azuredevops/resource_build_definition.go]: Can we add a `ValidateFunc` (perhaps `ValidateNoEmptyString` or similar -- i forget the actual name, but its used elsewhere in the project)

Comment (exocom) [reply azuredevops/resource_build_definition.go]: Good idea! I've added a validation function. The pattern comes straight from Azure DevOps website.

**Comment by nmiodice** [comment]
@exocom I'd suggest (for the testing) to do the following:
 - Add a step similar to [this](https://github.com/exocom/terraform-provider-azuredevops/blob/build_pipepine_path/azuredevops/resource_build_definition_test.go#L222-L230). You can update this code to take a definition path as an argument and change it in the last step
 - You can then add another check for the expected path, which could be inserted [here](https://github.com/exocom/terraform-provider-azuredevops/blob/build_pipepine_path/azuredevops/resource_build_definition_test.go#L273)

**Comment by exocom** [comment]
> @exocom I'd suggest (for the testing) to do the following:
> 
> * Add a step similar to [this](https://github.com/exocom/terraform-provider-azuredevops/blob/build_pipepine_path/azuredevops/resource_build_definition_test.go#L222-L230). You can update this code to take a definition path as an argument and change it in the last step
> * You can then add another check for the expected path, which could be inserted [here](https://github.com/exocom/terraform-provider-azuredevops/blob/build_pipepine_path/azuredevops/resource_build_definition_test.go#L273)

@nmiodice I've added the tests. Can you please review and let me know. Thanks for helping me out!

Also when the build fails I am unable to see the build log on Azure Pipelines I get a 401. Is there anyway for us to view the build log?

**Comment by nmiodice** [comment]
@exocom I'm hesitant to allow access to the build pipeline as it requires access to internal resources. Let me think about how to make this less painful for OSS contributors in the future :)

**Pull request merged**